lib/Service/BreakoutRoomService.php

```diff
@@ -0,0 +1,237 @@
+
+		$moderators = array_filter($participants, static fn (Participant $participant) => $participant->hasModeratorPermissions());
+		$this->addModeratorsToBreakoutRooms($breakoutRooms, $moderators);
+
+		$others = array_filter($participants, static fn (Participant $participant) => !$participant->hasModeratorPermissions());
+		if ($mode === BreakoutRoom::MODE_AUTOMATIC) {
+			// Shuffle the attendees, so they are not always distributed in the same way
+			shuffle($others);
+
+			$map = [];
+			foreach ($others as $index => $participant) {
+				$map[$index % $amount] ??= [];
+				$map[$index % $amount][] = $participant;
+			}
+
+			$this->addOthersToBreakoutRooms($breakoutRooms, $map);
+		} elseif ($mode === BreakoutRoom::MODE_MANUAL) {
+			$map = [];
+			foreach ($others as $participant) {
+				$roomNumber = $attendeeMap[$participant->getAttendee()->getId()] ?? null;
+				if ($roomNumber === null) {
+					continue;
+				}
+
+				$roomNumber = (int) $roomNumber;
+
+				$map[$roomNumber] ??= [];
+				$map[$roomNumber][] = $participant;
+			}
+
+			$this->addOthersToBreakoutRooms($breakoutRooms, $map);
+		}
+
+
+		return $breakoutRooms;
+	}
+
+	/**
+	 * @param Room[] $rooms
+	 * @param Participant[] $moderators
+	 */
+	protected function addModeratorsToBreakoutRooms(array $rooms, array $moderators): void {
+		$moderatorsToAdd = [];
+		foreach ($moderators as $moderator) {
+			$attendee = $moderator->getAttendee();
+
+			$moderatorsToAdd[] = [
+				'actorType' => $attendee->getActorType(),
+				'actorId' => $attendee->getActorId(),
+				'displayName' => $attendee->getDisplayName(),
+				'participantType' => $attendee->getParticipantType(),
+			];
+		}
+
+		foreach ($rooms as $room) {
+			$this->participantService->addUsers($room, $moderatorsToAdd);
+		}
+	}
+
+	/**
+	 * @param array $rooms
+	 * @param Participant[][] $participantsMap
+	 */
+	protected function addOthersToBreakoutRooms(array $rooms, array $participantsMap): void {
+		foreach ($rooms as $roomNumber => $room) {
+			$toAdd = [];
+
+			$participants = $participantsMap[$roomNumber] ?? [];
+			foreach ($participants as $participant) {
+				$attendee = $participant->getAttendee();
+
+				$toAdd[] = [
+					'actorType' => $attendee->getActorType(),
+					'actorId' => $attendee->getActorId(),
+					'displayName' => $attendee->getDisplayName(),
+					'participantType' => $attendee->getParticipantType(),
+				];
+			}
+
+			if (empty($toAdd)) {
+				continue;
+			}
+
+			$this->participantService->addUsers($room, $toAdd);
+		}
+	}
+
+	public function createBreakoutRooms(Room $parent, int $amount): array {
+		// Safety caution cleaning up potential orphan rooms
+		$this->deleteBreakoutRooms($parent);
+
+		$rooms = [];
+		for ($i = 1; $i <= $amount; $i++) {
+			$breakoutRoom = $this->roomService->createConversation(
+				$parent->getType(),
+				str_replace('{number}', (string) $i, $this->l->t('Room {number}')),
```

Comment:
Is good to put a hint to translators to make more easy to translators identify the place of this phrase will occur and get a best understanding about what is this phrase.

https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/l10n.html#provide-context-hints-for-translators

```suggestion
				// TRANSLATORS The room name followed with a sequential number from 1 to N where N is the total of breakout rooms.
				$this->l->t('Room %n', $i),
```

Reply:
We can not use `%n` as it is not a plural. We could use an ordinary `%s` placeholder but I find it better with "speaking" placeholders like in javascript.
But I will add a translator hint additionally

Reply:
This is a good point to follow up to server and change the `t` function to accept the same syntax as JS.